template.yaml

```diff
@@ -37,17 +58,142 @@ Resources:
               - sts:AssumeRole
       ManagedPolicyArns:
         - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+        - !Ref ALBListenerRuleFunctionPolicy
+        - !Ref KmsDecryptManagedPolicy
+        - !Ref SsmManagedPolicy
+
+  ALBListenerRuleFunctionPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: DoElbStuff
+            Effect: 'Allow'
+            Action:
+              - 'elasticloadbalancing:*'
+            Resource: '*'
+
+  KmsDecryptManagedPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: ReadAccess
+            Action:
+              - 'kms:ListKeys'
+              - 'kms:ListAliases'
+              - 'kms:DescribeKey'
+              - 'kms:ListKeyPolicies'
+              - 'kms:GetKeyPolicy'
+              - 'kms:GetKeyRotationStatus'
+              - 'iam:ListUsers'
+              - 'iam:ListRoles'
+            Effect: Allow
+            Resource: !GetAtt KmsKey.Arn
+          - Sid: DecryptAccess
+            Action:
+              - 'kms:Encrypt'
+              - 'kms:Decrypt'
+            Effect: Allow
+            Resource: !GetAtt KmsKey.Arn
+
+  SsmManagedPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: GetParamAccess
+            Action:
+              - 'ssm:*'
+            Effect: Allow
+            Resource: !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/alb-notebook-access/AuthenticateOidcClientSecret'
```

Comment:
I assume we will manually add this secret to the SSM using the `KmsKey`?

Reply:
Maybe store all the OIDC parameters in the same SSM path?

Reply:
@zaro0508 yes, manually added. Already done in dev account, needs to be done in prod.
@ahayden I don't see the utility of doing that when those values don't need to be secret.